integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/ViewPopRedirectTests/RedirectTest.java

```diff
@@ -0,0 +1,132 @@
+     * 3. Call the popandRedirect fucntion and verify that browseMessage function is returning is expected message
+     */
+
+    @Test(groups = {"wso2.esb"}, description = "Test Redirect service for Message processor")
+    public void testRedirect() throws Exception {
+
+        //Initializing Payload
+        String inputPayload = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                + "<soapenv:Header/>\n"
+                + "<soapenv:Body>\n"
+                + "<m0:getQuote xmlns:m0=\"http://services.samples\">\n"
+                + " <m0:request>IBM\n"
+                + " </m0:request>\n"
+                + "   <m0:request>WSO2\n"
+                + " </m0:request>\n"
+                + "</m0:getQuote>\n"
+                + "</soapenv:Body>\n"
+                + "</soapenv:Envelope>";
+
+        String expectedMessage = "<?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">"
+                + "<soapenv:Body>"
+                + "<m0:getQuote xmlns:m0=\"http://services.samples\">\n"
+                + " <m0:request>IBM\n"
+                + " </m0:request>\n"
+                + "   <m0:request>WSO2\n"
+                + " </m0:request>\n"
+                + "</m0:getQuote>"
+                + "</soapenv:Body>"
+                + "</soapenv:Envelope>";
+
+        Map<String, String> requestHeader = new HashMap<>();
+        requestHeader.put("Content-type", "text/xml");
+        requestHeader.put("SOAPAction", "urn:mediate");
+        requestHeader.put("Accept", "application/json");
+
+        //Send payload to the proxy
+        System.out.println("=== Sending payload to " + PROXY_NAME + ":" + getProxyServiceURLHttp(PROXY_NAME));
+        HttpRequestUtil.doPost(new URL(getProxyServiceURLHttp(PROXY_NAME)), inputPayload, requestHeader);
+
+        Thread.sleep(5000); //Make sure that the Message Processor has time to deactivate
+
+        //Check if the message processor has deactivated
+        Assert.assertFalse(messageProcessorClient.isActive(PROCESSOR_NAME), "Message processor" + PROCESSOR_NAME + "should not be active, " +
+                "but it is active.");
+
+        //Deactivate the REDIRECT_PROCESSOR
+        messageProcessorClient.deactivateProcessor(REDIRECT_PROCESSOR_NAME);
+        Assert.assertFalse(messageProcessorClient.isActive(REDIRECT_PROCESSOR_NAME), "Message processor" + REDIRECT_PROCESSOR_NAME + " should not be active, " +
+                "but it is active.");
+
+        //Call popAndRedirect function to pass the message to REDIRECT_STORE
+        messageProcessorClient.popAndRedirectMessage(PROCESSOR_NAME, REDIRECT_STORE_NAME);
+
+
+        //Call the getMessage function from the REDIRECT_PROCESSOR
+        String returnedMessage = messageProcessorClient.browseMessage(REDIRECT_PROCESSOR_NAME);
+        System.out.println("=== RETURNED MESSAGE === \n" + returnedMessage);
```

Comment:
Shall we use `log` instead of system outs?

Reply:
The system outs have been removed in a different commit. It does not show up here, but the current Tests do not have system outs

Reply:
https://github.com/wso2/product-ei/blob/master/integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/ViewPopRedirectTests/RedirectTest.java